library/core/src/ptr/non_null.rs

```diff
@@ -1826,7 +1858,68 @@ mod verify {
     pub fn non_null_check_new() {
         let mut x: i32 = kani::any();
         let xptr = &mut x;
-        let maybe_null_ptr =  if kani::any() { xptr as *mut i32 } else { null_mut() };
+        let maybe_null_ptr = if kani::any() { xptr as *mut i32 } else { null_mut() };
         let _ = NonNull::new(maybe_null_ptr);
     }
+
+    #[kani::proof_for_contract(NonNull::<T>::copy_to)]
+    pub fn non_null_check_copy_to() {
+        // PointerGenerator instance
+        let mut generator = PointerGenerator::<16>::new();
```

Comment:
How the capacity of the generator affect the running time of the proof? Can we use the maximum capacity, e.g., `isize::MAX / size_of::<i32>()` here?

Reply:
With size 16 the runtime is about one second:
```
Checking harness ptr::non_null::verify::non_null_check_copy_to...

VERIFICATION RESULT:
 ** 0 of 328 failed (4 unreachable)

VERIFICATION:- SUCCESSFUL
Verification Time: 0.9928743s
```
With `usize::MAX / size_of::<i32>()` the value is too big:
```
error: internal compiler error: kani-compiler/src/codegen_cprover_gotoc/context/goto_ctx.rs:336:9: failed to get layout for `kani::arbitrary_ptr::PointerGenerator<4611686018427387903>`: values of the type `[u8; 4611686018427387903]` are too big for the target architecture
```
And with `isize::MAX as usize)/ size_of::<i32>()` the runtime is similar but it resulted in four failed checks:
```
Checking harness ptr::non_null::verify::non_null_check_copy_to...

VERIFICATION RESULT:
 ** 4 of 328 failed (4 unreachable)
Failed Checks: CAR size is less than __CPROVER_max_malloc_size
 File: "<builtin-library-__CPROVER_contracts_library>", line 792, in __CPROVER_contracts_write_set_check_assignment
Failed Checks: no offset bits overflow on CAR upper bound computation
 File: "<builtin-library-__CPROVER_contracts_library>", line 798, in __CPROVER_contracts_write_set_check_assignment
Failed Checks: CAR size is less than __CPROVER_max_malloc_size
 File: "<builtin-library-__CPROVER_contracts_library>", line 164, in __CPROVER_contracts_car_set_insert
Failed Checks: no offset bits overflow on CAR upper bound computation
 File: "<builtin-library-__CPROVER_contracts_library>", line 168, in __CPROVER_contracts_car_set_insert

VERIFICATION:- FAILED
Verification Time: 1.2587996s
```